sigstore/newtypes.py

```diff
@@ -0,0 +1,6 @@
+from typing import NewType
```

Comment:
My preference would be to avoid a new module here (especially a public one) -- IMO these newtypes should go in the existing `_utils` module, since that's where a lot of them get constructed anyways.

Reply:
Should we then `import _utils` instead of `import newtypes` in every file that needs changes? I am wondering if there is a way of putting all the needed newtypes in a function and doing `from _utils import function` instead.

Reply:
> Should we then import _utils instead of import newtypes in every file that needs changes? 

Yes, exactly (although you should use the fully qualified `sigstore._utils` path for consistency with the rest of the code).

> I am wondering if there is a way of putting all the needed newtypes in a function and doing from _utils import function instead.

This is probably overkill -- having them at the top-level of `_utils` is idiomatic, and we can selectively import them as needed.